packages/app/src/studio/StudioPanel.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <div v-if="error">
+    Error loading the panel
+  </div>
+  <div
+    v-else
+    ref="root"
+  >
+    Loading the panel...
+  </div>
+</template>
+<script lang="ts" setup>
+import { ref, onMounted, onUpdated, onBeforeUnmount } from 'vue'
+import { init, loadRemote } from '@module-federation/runtime'
+
+const root = ref<HTMLElement | null>(null)
+const error = ref<string | null>(null)
+const Panel = ref<ReturnType<any> | null>(null)
+
+const updateComponent = () => {
+  if (!Panel.value || !!error.value) {
+    return
+  }
+
+  const panel = window.UnifiedRunner.React.createElement(Panel.value)
+
+  window.UnifiedRunner.ReactDOM.createRoot(root.value).render(panel)
+}
+
+const unmountReactComponent = () => {
+  if (!Panel.value || !root.value) {
+    return
+  }
+
+  window.UnifiedRunner.ReactDOM.unmountComponentAtNode(root.value)
+}
+
+init({
+  remotes: [{
+    alias: 'app-studio',
+    type: 'module',
+    name: 'app-studio',
+    entryGlobalName: 'app-studio',
+    entry: '/__cypress-studio/app-studio.js',
+    shareScope: 'default',
+  }],
+  shared: {
+    react: {
+      scope: 'default',
+      version: '18.3.1',
+      lib: () => window.UnifiedRunner.React,
+      shareConfig: {
+        singleton: true,
+        requiredVersion: '^18.3.1',
+      },
+    },
+  },
+  name: 'app',
+})
+
+onMounted(updateComponent)
+onUpdated(updateComponent)
```

Comment:
when this hook is called does it remove the previous component? Almost looks like it would create a duplicate if I am reading it correctly? 


Or does it override the node in the dom with a new instance? do we need to destroy the previous root if its detached?

Reply:
Doh. I also meant to remove this as we don't have a need to re-render things in this way. Removed here and also cleaned up another lifecycle bug: [`6fb9d30` (#31129)](https://github.com/cypress-io/cypress/pull/31129/commits/6fb9d30536f2a5e1c9575a5b030525a1050e36ef)